cmd/mailer/internal/services/scheduler.go

```diff
@@ -0,0 +1,53 @@
+package services
+
+import (
+	"time"
+
+	"github.com/fdemchenko/exchanger/internal/communication"
+	"github.com/fdemchenko/exchanger/internal/communication/mailer"
+	"github.com/fdemchenko/exchanger/internal/communication/rabbitmq"
+	"github.com/rs/zerolog/log"
+)
+
+type EmailScheduler struct {
+	producer *rabbitmq.GenericProducer
+	stopChan chan bool
+}
+
+func NewEmailScheduler(
+	producer *rabbitmq.GenericProducer,
+) *EmailScheduler {
+	return &EmailScheduler{
+		stopChan: make(chan bool),
+		producer: producer,
+	}
+}
+
+func (es *EmailScheduler) StartBackhroundTask(updateInterval time.Duration) {
+	ticker := time.NewTicker(updateInterval)
```

Comment:
Most likely I've missed such implementation of scheduled action in previous PRs. However I've to notice it's not the most optimal way to implement. It's hard to control when messages will be sent cause it depends on start time of the mailer service. The only what is controlled with current implementation is frequency, but not absolute time of execution.

Reply:
Yeah, agree. It's time to switch to CRON scheduled task